python/ray/serve/backend_state.py

```diff
@@ -606,26 +570,82 @@ def __init__(self, name: str, controller_name: str, detached: bool,
         self._target_replicas: int = -1
         self._curr_goal: Optional[GoalId] = None
         self._target_version: BackendVersion = None
-        self._prev_startup_warning: float = -1
+        self._prev_startup_warning: float = time.time()
         self._replica_constructor_retry_counter: int = 0
         self._replicas: ReplicaStateContainer = ReplicaStateContainer()
 
-    def get_checkpoint_data(self):
-        return (self._target_info, self._rollback_info, self._target_replicas,
-                self._curr_goal, self._target_version,
+    def get_target_state_checkpoint_data(self):
+        """
+        Return deployment's target state submitted by user's deployment call.
+        Should be persisted and outlive current ray cluster.
+        """
+        return (self._target_info, self._target_replicas, self._target_version)
+
+    def get_current_state_checkpoint_data(self):
+        """
+        Return deployment's current state specific to the ray cluster it's
+        running in. Might be lost or re-constructed upon ray cluster failure.
+        """
+        return (self._rollback_info, self._curr_goal,
```

Comment:
why is rollback info in this and not target state?

Reply:
I think self._rollback_info is somewhat specific to current state since it will only existing while we have a pending deploy() operation, also it's not part of BackendInfo that we unpack and persist as part of _set_backend_goal(). Either way can work tho since it's pickled and have nothing to do with actor names